### ISSUE TITLE
Project dependencies may have API risk issues

### ISSUE THREAD

**Issue by PyDeps** [kuprel/min-dalle#96]
Hi, In **min-dalle**, inappropriate dependency versioning constraints can cause risks.

Below are the dependencies and version constraints that the project is using

```
min-dalle
numpy==1.23.0
pillow==9.2.0
requests==2.28.1
```

The version constraint **==** will introduce the risk of dependency conflicts because the scope of dependencies is too strict.
The version constraint **No Upper Bound** and **\*** will introduce the risk of the missing API Error because the latest version of the dependencies may remove some APIs.

After further analysis, in this project,
The version constraint of dependency **pillow** can be changed to *==9.2.0*.
The version constraint of dependency **pillow** can be changed to *>=2.0.0,<=9.1.1*.
The version constraint of dependency **requests** can be changed to *>=0.2.1,<=0.2.3*.
The version constraint of dependency **requests** can be changed to *>=0.7.0,<=2.24.0*.
The version constraint of dependency **requests** can be changed to *==2.26.0*.

The above modification suggestions can reduce the dependency conflicts as much as possible, 
and introduce the latest version as much as possible without calling Error in the projects.

The invocation of the current project includes all the following methods.

<details><summary>The calling methods from the pillow</summary>
 <pre>PIL.Image.new
PIL.Image.fromarray
PIL.ImageTk.PhotoImage
</pre>
</details>
<details><summary>The calling methods from the requests</summary>
 <pre>requests.get
</pre>
</details>
<details><summary>The calling methods from the all methods</summary>
 <pre>math.sqrt
torch.einsum
self.device.self.layer_count.self.glu_embed_count.self.embed_count.self.attention_head_count.self.image_vocab_count.DalleBartDecoder.to
ord
f.read
self.lm_head.sort
j.self.block.forward
argparse.ArgumentParser.parse_args
self.v.forward
size.int.size.image.resize.convert
len
tkinter.StringVar.get
models.VQGanDetokenizer
self.k.forward
ResnetBlock
self.conv.forward.to
model.generate_image.resize
self.layernorm_embedding.forward
self.conv2
size.int.size.image.resize.convert.getdata
tkinter.BooleanVar.get
os.cpu_count
os.path.join.endswith
self.out_proj.forward.reshape
self.q.forward
self.download_tokenizer
self.download_encoder
self.get_byte_pair_encoding
set_mega_false_and_destroy.frm.ttk.Button.grid
list.index
image.reshape.reshape
self.norm_out.forward
torch.nn.Linear
self.encoder.load_state_dict
tkinter.BooleanVar
Decoder
Upsample
PIL.ImageTk.PhotoImage
torch.nn.GELU
super
set_mega_true_and_destroy.frm.ttk.Button.grid
super.forward
self.q_proj.forward.reshape
padding_image.frm.ttk.Label.grid
save_image
regen_root
models.VQGanDetokenizer.eval
self.encoder_attn_layer_norm.forward
z.flatten.transpose.flatten.permute
z.flatten.transpose.flatten.clip
self.embedding.forward
sv_topk.props.ttk.Entry.grid
model.generate_image.save
torch.stack
zip
DecoderSelfAttention
f.read.split
j.self.attn.forward
tkinter.ttk.Frame.grid
os.makedirs
self.pre_self_attn_layer_norm.forward
self.generate_image_stream
self.glu.forward
i.self.up.forward
self.init_decoder
settings.to
os.path.join
image.reshape.transpose
self.lm_head
print
self.lm_head.exp_
torch.load
token_index.unsqueeze.repeat
self.mid.forward
self.generate_raw_image_stream
w.permute.permute
tkinter.ttk.Checkbutton
text_tokenizer.TextTokenizer
torch.tensor.not_equal
torch.arange
v.reshape.reshape
i.self.layers.forward
self.embed_positions.forward
torch.manual_seed
tkinter.Tk.mainloop
self.encoder.forward
text.lower.encode.decode.lower
self.v_proj.forward
self.image_grid_from_tokens
torch.cuda.empty_cache
self.k_proj.forward
torch.nn.ModuleList
torch.bmm
self.conv_out.forward
self.token_from_subword.get
self.init_tokenizer
sys.exit
self.ln0.forward
PIL.Image.fromarray
torch.cuda.is_available
next
format
self.dtype.self.device.self.layer_count.self.glu_embed_count.self.embed_count.self.attention_head_count.self.image_vocab_count.DalleBartDecoder.to.eval
self.download_detokenizer
self.conv1.forward
range
tkinter.ttk.Frame
tkinter.Tk.destroy
z.flatten.unsqueeze
join
sv_prompt.props.ttk.Entry.grid
self.block_1.forward
generate_image
torch.softmax
tkinter.Tk
self.detokenizer.forward
self.generate_images_stream
torch.zeros
z.flatten.transpose
json.load
super.__init__
values.keys.torch.cat.to
image.reshape.to
prev_tokens.repeat.repeat
self.gelu.forward
argparse.ArgumentParser
setuptools.setup
text.strip.split
self.detokenizer.load_state_dict
argparse.ArgumentParser.add_argument
self.out_proj.forward
layer.forward
self.decoder.to
self.model.generate_image_stream
self.tokenizer.tokenize
str
numpy.ones
final_image.save
torch.uint8.image.to.to.numpy
self.final_ln
self.conv.forward
os.path.isdir
models.DalleBartEncoder
self.norm
torch.nn.UpsamplingNearest2d
torch.nn.LayerNorm
z.flatten.transpose.flatten
tkinter.ttk.Label.grid
tkinter.ttk.Entry
emoji.demojize
self.norm2.forward
self.block_2.forward
float
open
torch.cuda.amp.autocast
DecoderCrossAttention
torch.nn.GroupNorm
root.destroy.props.ttk.Button.grid
f.write
self.fc0.forward
image.reshape.view
self.conv_in.forward
self.detokenizer.to
q.permute.reshape
props.ttk.Label.grid
torch.full
self.proj_out.forward
filename_from_text
UpsampleBlock
text.lower.encode
tkinter.ttk.Label
EncoderSelfAttention
ascii_from_image
self.q_proj.forward
min
dalle_bart_encoder.GLU
min_dalle.MinDalle.generate_image_stream
PIL.Image.new
self.embed_tokens.forward
q.permute.permute
image.save
DecoderLayer
torch.cat
min_dalle.MinDalle.generate_image
text_tokens.not_equal.to
self.post_quant_conv.forward
text.lower.encode.decode
self.forward
self.k_proj.forward.reshape
tkinter.ttk.Button
self.upsample.forward.reshape
tkinter.ttk.Label.configure
bool
text.lower.encode.decode.split
save.props.ttk.Button.grid
tkinter.StringVar
torch.nn.Embedding
min_dalle.MinDalle
join.strip
torch.set_grad_enabled
tkinter.StringVar.set
torch.nn.Conv2d
self.decoder.forward
tkinter.ttk.Label.update
tkinter.Tk.wm_resizable
argparse.ArgumentParser.set_defaults
padding_image.props.ttk.Label.grid
sv_temperature.props.ttk.Entry.grid
is_kept.to
self.device.self.layer_count.self.text_vocab_count.self.text_token_count.self.glu_embed_count.self.embed_count.self.attention_head_count.DalleBartEncoder.to
self.download_decoder
join.lower
self.pre_encoder_attn_layer_norm.forward
os.path.exists
self.fc1.forward
self.norm1.forward
torch.sigmoid
requests.get
z.flatten.transpose.flatten.flatten
text_tokens.not_equal
self.init_detokenizer
self.decoder.load_state_dict
token_index.unsqueeze.repeat.unsqueeze
self.self_attn_layer_norm.forward
self.final_ln.forward
EncoderLayer
self.fc2.forward
k.reshape.reshape
MiddleLayer
self.encoder.to
pair.split
self.nin_shortcut.forward
sv_supercond.props.ttk.Entry.grid
dict
torch.set_num_threads
torch.uint8.image.to.to
self.v_proj.forward.reshape
self.encoder_attn.forward
cog.Path
list
chr
self.ln1.forward
self.dtype.self.device.self.layer_count.self.text_vocab_count.self.text_token_count.self.glu_embed_count.self.embed_count.self.attention_head_count.DalleBartEncoder.to.eval
tempfile.mkdtemp
self.rank_from_pair.get
self.attn_1.forward
models.DalleBartDecoder
generate.props.ttk.Button.grid
int
bv_seamless.props.ttk.Checkbutton.grid
GLU
AttentionBlock
reversed
self.decoder.sample_tokens
cog.Input
self.self_attn.forward
torch.multinomial
tuple
self.upsample.forward
z.flatten.transpose.flatten.view
z.permute.contiguous
torch.tensor
self.init_encoder
</pre>
</details>

@developer
Could please help me check this issue?
May I pull a request to fix it?
Thank you very much.
